lib/beaker-pe/install/pe_utils.rb

```diff
@@ -365,6 +367,92 @@ def deploy_frictionless_to_master(host)
+          when :simple_monolithic
+            simple_monolithic_install(hosts.first, hosts.drop(1), opts)
+          when :simple_split
+            # This isn't implemented yet, so just do a generic install instead
+            #simple_split_install(hosts, opts)
+            generic_install(hosts, opts)
+          else
+            generic_install(hosts, opts)
+          end
+        end
+
+        def has_all_roles?(host, roles)
+          roles.all? {|role| host['roles'].include?(role)}
+        end
+
+        def determine_install_type(hosts, opts)
+          # Do a generic install if this is masterless, not all the same PE version, an upgrade, or earlier than 2016.4
+          return :generic if opts[:masterless]
+          return :generic if hosts.map {|host| host['pe_ver']}.uniq.length > 1
+          return :generic if opts[:type] == :upgrade
+          return :generic if version_is_less(opts[:pe_ver] || hosts.first['pe_ver'], '2016.4')
+
+          mono_roles = ['master', 'database', 'dashboard']
+          if has_all_roles?(hosts.first, mono_roles) && hosts.drop(1).all? {|host| host['roles'].include?('frictionless')}
+            :simple_monolithic
+          elsif hosts[0]['roles'].include?('master') && hosts[1]['roles'].include?('database') && hosts[2]['roles'].include?('dashboard') && hosts.drop(3).all? {|host| host['roles'].include?('frictionless')}
+            :simple_split
+          else
+            :generic
+          end
+        end
+
+        def simple_monolithic_install(master, agents, opts={})
+          step "Performing a standard monolithic install with frictionless agents"
+          all_hosts = [master, *agents]
+          configure_type_defaults_on(all_hosts)
+
+          # Set PE distribution on the master, create working dir
+          prepare_hosts([master], opts)
+          fetch_pe([master], opts)
+          prepare_host_installer_options(master)
+          generate_installer_conf_file_for(master, [master], opts)
+          on master, installer_cmd(master, opts)
+
+          step "Setup frictionless installer on the master" do
+            agents.each do |agent|
+              # If We're *not* running the classic installer, we want
+              # to make sure the master has packages for us.
+              if agent['platform'] != master['platform'] # only need to do this if platform differs
+                deploy_frictionless_to_master(agent)
+              end
+            end
+          end
+
+          step "Install agents" do
+            agents.group_by {|agent| installer_cmd(agent, opts)}.each do |cmd, agents|
+              on agents, cmd, :run_in_parallel => true
+            end
+          end
+
+          step "Stop puppet agents to avoid interfering with tests" do
+            stop_agent_on(all_hosts, :run_in_parallel => true)
+          end
+
+          step "Sign agent certificates" do
+            # This will sign all cert requests
+            sign_certificate_for()
```

Comment:
Oh, looks like `sign_certificate_for(hosts)` will still try to sign them all at once, and will repeat until they're all signed. Will update it to just do that then!